prov/verbs/src/ep_rdm/verbs_rdm.h

```diff
@@ -307,23 +316,16 @@ struct fi_ibv_rdm_tagged_postponed_entry {
 	struct fi_ibv_rdm_tagged_conn *conn;
 };
 
-enum fi_ibv_rdm_buffer_status {
-	BUF_STATUS_FREE = 0,
-	BUF_STATUS_BUSY
-};
-
+/*
+ * Size of the structure is performance sensitive. TODO: resize to 4 bytes.
+ */
 struct fi_ibv_rdm_buffer_service_data {
 	volatile enum fi_ibv_rdm_buffer_status status;
```

Comment:
> > -enum fi_ibv_rdm_buffer_status {
> > -	BUF_STATUS_FREE = 0,
> > -	BUF_STATUS_BUSY
> > -};
> > -
> > +/*
> > + * Size of the structure is performance sensitive. TODO: resize to 4
> bytes.
> > + */
> >  struct fi_ibv_rdm_buffer_service_data {
> >  	volatile enum fi_ibv_rdm_buffer_status status;
> 
> Buffer's status is a part of buffers control. Remote side is updating this
> address by RDMA_WRITE operation to release then it's can be reused.

Got it -- then my concern is that this eventually should handle endian issues.  They way I've handled endian before is to exchange he endian format as part of connection setup and do byte swapping at the initiator if the endian formats do not match.

To do this, you just need to make sure that there's a reserved flag available during connection data exchange, then the feature can be enabled later.